src/index.js

```diff
@@ -313,145 +289,174 @@ function addSegment(currentState, segment) {
 
 // The main interface
 
-export var RouteRecognizer = function() {
-  this.rootState = new State();
-  this.names = {};
-};
-
-
-RouteRecognizer.prototype = {
-  add: function(routes, options) {
-    var currentState = this.rootState, regex = "^",
-        types = { statics: 0, dynamics: 0, stars: 0 },
-        handlers = [], allSegments = [], name;
-
-    var isEmpty = true;
-
-    for (var i=0, l=routes.length; i<l; i++) {
-      var route = routes[i], names = [];
-
-      var segments = parse(route.path, names, types);
+export class RouteRecognizer {
+  constructor() {
+    this.map = map;
```

Comment:
Need to figure out who uses this. It seems unused, but aurelia/router doesn't have enough tests for me to confidently delete it. Need to look closer.